### PR TITLE
Updated for Dart 2

### PR DESCRIPTION
**Pull request by AppleEducate**

### CODE REVIEW

**Comment by AppleEducate** [comment]
Fixed methods and variable names for Dart 2 compatibility.

**Comment by AppleEducate** [comment]
@osteele how does this look?

File: lib/src/chord.dart [hunk 1]
```diff
@@ -43,36 +47,44 @@ class ChordPattern {
...
+        'intervals': intervals
+      }.toString();
 
   static bool _chordsInitialized = false;
 
   static void _initializeChords() {
     if (_chordsInitialized) return;
     _chordsInitialized = true;
-    for (var spec in ChordPatternSpecs) {
+    for (var spec in cordPatternSpecs) {
```
Comment (osteele) [inline lib/src/chord.dart]: `cordPatternSpecs` → `chordPatternSpecs`

File: lib/src/fretting.dart [hunk 2]
```diff
@@ -126,42 +142,62 @@ class FretPosition {
...
   });
   return positions;
 }
 
-List<Fretting> chordFrettings(Chord chord, FrettedInstrument instrument, {int highestFret: 4}) {
+List<Fretting> chordFrettings(Chord chord, FrettedInstrumentClass instrument,
+    {highestFret: 4}) {
   var warn = false;
   int minPitchClasses = chord.intervals.length;
+  print(warn);
```
Comment (osteele) [inline lib/src/fretting.dart]: Delete `print(warn);`?

Comment (AppleEducate) [reply lib/src/fretting.dart]: The only reason I added the `print(warn);` was because it wasn’t being used and I was getting rid of the dart warning.

**Comment by osteele** [comment]
Thank you for your contribution!

`.packages` and `.vscode/launch.json` shouldn't be added. Instead, add `.packages` and `.vscode/` to this project's `.gitignore`, or your global Git ignore if you have one. (It would be reasonable to add them both to `.gitignore`, since `.packages` is in the standard [Dart gitignore template](https://github.com/dvcs/gitignore/blob/master/templates/Dart.gitignore), and since VS Code is so popular.)

`Instrument` → `InstrumentClass`, `FrettedInstrument` → `FrettedInstrument` — not sure about the name change. (I'm not saying it's wrong, just that the motivation isn't evident to me.) Walk me through the rationale?

Renaming `Interval.M3` → `Interval.iM3` — I appreciate that you've done a lot of work to bring this up to the Dart 2.0 guidelines for constant names, but I wonder about the result. This might be a case where, when coding in the domain of music theory, it makes more sense to invoke the “In existing code that uses SCREAMING_CAPS, you may continue to use all caps to stay consistent.” clause of <http://dartdoc.takyam.com/articles/style-guide/#names> and <https://www.dartlang.org/guides/language/effective-dart/style#prefer-using-lowercamelcase-for-constant-names>, as [dart:html has done](https://github.com/dart-lang/sdk/issues/31367).

On the other hand, I'm not active in programming in Dart and don't have a sense of what looks most readable in practice, so in the end I'd like to defer to your judgement and recommendation. I just wanted to raise this as a question.

**Comment by AppleEducate** [comment]
Absolutely! And I had to change the name classes because there were conflicts with default naming conventions in flutter. For example Instrument is already talked by the Instrement Test Cases class for flutter. Is there a better way to deal with naming conflicts? 

and for the iM3 i thought it was going to be deperecated and so i changed it because of all the warnings in dart 2. But if they can stay that would be even better. 

I actively use Dart 2 for flutter and not for the web, so a lot of what i did reflected this. I actually am using the version I am merging in an app already. It works really good and this is the only music theory package for flutter. 

I will do the .gitignore stuff too.

**Comment by osteele** [comment]
Re `Instrument` vs.`InstrumentClass`:[Specifying a library prefix](https://www.dartlang.org/guides/language/language-tour#specifying-a-library-prefix) looks like the general Dart solution for naming conflicts. Does that help? What package defines the Instrument Test Cases for flutter (I didn't see anything in https://github.com/flutter/flutter/tree/master/packages/flutter_driver or https://github.com/flutter/flutter/blob/master/packages/flutter_test), and is this imported explicitly or through some test framework magic?

Re `M3`/`iM3` etc.: Let's stay with `M3`. Does this trigger a lint warning? It looks like this is rule `constant_identifier_names` of [dart-lang/linter](https://github.com/dart-lang/linter) (used by `dartanalyzer`), but this isn't in [Google's own settings](https://github.com/dart-lang/linter/blob/master/example/google.yaml), and interestingly when I tried enabling this it didn't actually complain about the code that's currently in `master`.

Glad to hear you're finding this useful!

**Comment by AppleEducate** [comment]
@osteele  ok so I changed some of the names back. The problems were showing up in vscode, so I don't think it is from the analyzer or lint. And forgot about the library prefix, that would work great.

**Comment by AppleEducate** [comment]
For reference, this is from `flutter analyze`

```
Analyzing dart-tonic...

   info • The method '_defaultQualityForNumber' isn't used • lib/src/interval.dart:56:17
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:145:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:147:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:149:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:150:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:151:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:152:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:154:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:156:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:157:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:159:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:160:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:161:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:162:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:163:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:164:25
   info • Name non-constant identifiers using lowerCamelCase • lib/src/interval.dart:165:25
   info • Unused import: 'tonic_test.dart' • test/interval_test.dart:3:8
   info • Unused import: 'tonic_test.dart' • test/matchers.dart:3:8
   info • Unused import: 'tonic_test.dart' • test/pitch_class_test.dart:3:8
   info • Unused import: 'tonic_test.dart' • test/pitch_test.dart:3:8
   info • Unused import: 'tonic_test.dart' • test/scales_test.dart:3:8
   info • Unused import: 'chord_test.dart' • test/tonic_test.dart:3:8
   info • Unused import: 'fretting_test.dart' • test/tonic_test.dart:4:8
   info • Unused import: 'instrument_test.dart' • test/tonic_test.dart:5:8
   info • Unused import: 'scales_test.dart' • test/tonic_test.dart:6:8
   info • Unused import: 'pitch_class_test.dart' • test/tonic_test.dart:7:8
   info • Unused import: 'interval_test.dart' • test/tonic_test.dart:8:8
   info • Unused import: 'pitch_test.dart' • test/tonic_test.dart:9:8

29 issues found. (ran in 1.7s)
```

File: LICENSE [hunk 3]
```diff
@@ -1,3 +1,4 @@
+TODO: Add your license here.
```
Comment (osteele) [inline LICENSE]: remove TODO line

File: dart_tonic.iml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (osteele) [inline dart_tonic.iml]: Remove this file, and add `*.iml` to `.gitignore`.

Source: https://www.dartlang.org/guides/libraries/private-files

File: flutter_01.log [hunk 5]
```diff
@@ -0,0 +1,99 @@
+Flutter crash report; please file at https://github.com/flutter/flutter/issues.
```
Comment (osteele) [inline flutter_01.log]: Remove, and add `*.log` to `.gitignore`

File: lib/tonic.dart [hunk 6]
```diff
@@ -10,3 +10,9 @@ part 'src/scale.dart';
 part 'src/chord.dart';
 part 'src/instrument.dart';
 part 'src/fretting.dart';
+
+/// A Calculator.
+class Calculator {
```
Comment (osteele) [inline lib/tonic.dart]: ?

File: test/tonic_test.dart [hunk 7]
```diff
@@ -1,23 +1,39 @@
...
 
-part 'matchers.dart';
-part 'interval_test.dart';
-part 'pitch_class_test.dart';
-part 'pitch_test.dart';
-part 'scales_test.dart';
-part 'chord_test.dart';
-part 'instrument_test.dart';
-part 'fretting_test.dart';
+// import 'package:unittest/unittest.dart';
```
Comment (osteele) [inline test/tonic_test.dart]: ?

File: test/tonic_test.dart [hunk 8]
```diff
@@ -1,23 +1,39 @@
...
+// import 'package:tonic/tonic.dart';
 
-main() {
-  defineIntervalTests();
```
Comment (osteele) [inline test/tonic_test.dart]: ?

**Comment by AppleEducate** [comment]
Sorry I’ll update that, I created a new project and imported the old one to be up to date with the flutter sdk default project structure

**Comment by osteele** [comment]
Ah, `flutter analyze` uses different defaults from `dart analyze`.

Creating an empty file named `analysis_options.yaml` at the project root eliminates the "Name non-constant identifiers using lowerCamelCase" warnings.

**Comment by AppleEducate** [comment]
Ok fixed!

File: example/tonic_example.dart [hunk 9]
```diff
@@ -43,32 +45,36 @@ main() {
   print(Pitch.parse('C4') + Interval.d4); // => F♭4
   print(Pitch.parse('C4') + Interval.P4); // => F4
 
-  print(Pitch.parse('C4') - Pitch.parse('C4'));   // => P1
-  print(Pitch.parse('D4') - Pitch.parse('C4'));   // => M2
-  print(Pitch.parse('E4') - Pitch.parse('C4'));   // => M3
+  print(Pitch.parse('C4') - Pitch.parse('C4')); // => P1
+  print(Pitch.parse('D4') - Pitch.parse('C4')); // => M2
+  print(Pitch.parse('E4') - Pitch.parse('C4')); // => iM3
```
Comment (osteele) [inline example/tonic_example.dart]: iM3 → M3

File: example/tonic_example.dart [hunk 10]
```diff
@@ -43,32 +45,36 @@ main() {
...
   print(Pitch.parse('E♯4') - Pitch.parse('C4')); // => A3
   print(Pitch.parse('F♭4') - Pitch.parse('C4')); // => d4
-  print(Pitch.parse('F4') - Pitch.parse('C4'));  // => P4
+  print(Pitch.parse('F4') - Pitch.parse('C4')); // => iP4
```
Comment (osteele) [inline example/tonic_example.dart]: iP4 → P4

File: lib/src/instrument.dart [hunk 11]
```diff
@@ -18,15 +18,23 @@ class Instrument {
...
     _initialized = true;
-    for (var spec in _INSTRUMENT_SPECS) {
-      var stringPitches = spec['stringPitches'].split(new RegExp(r'\s+')).map(Pitch.parse).toList();
+    for (var spec in _instrumentSpecs) {
+      // var stringPitches = spec['stringPitches']
+      //     .split(new RegExp(r'\s+'))
+      //     .map(Pitch.parse)
+      //     .toList();
+      String _spec = spec['stringPitches'];
+      _spec.split(new RegExp(r'\s+')).map(Pitch.parse).toList();
```
Comment (osteele) [inline lib/src/instrument.dart]: it looks like these two lines are unnecessary

**Comment by osteele** [comment]
Almost there!

**Comment by AppleEducate** [comment]
Sweet! FIxed.

**Comment by AppleEducate** [comment]
Awesome! Will this be published to dart pub too?

**Comment by AppleEducate** [comment]
Gotcha, that’s for the quick feedback. 👍🏻

**Pull request merged**